### PR TITLE
Apply IndirectCallRewriter changes done in analysis-development branch

### PR DESCRIPTION
**Pull request by ptomin**
Apply `IndirectCallRewriter` changes done in #329 PR
Now it produces nice output for `VCExeSample`

### CODE REVIEW

File: src/Decompiler/Analysis/DataFlowAnalysis.cs [hunk 1]
```diff
@@ -87,17 +87,22 @@ public void BuildExpressionTrees()
...
 
-                    var icrw = new IndirectCallRewriter(program, ssa, eventListener);
-                    icrw.Rewrite();
+                    var vp = new ValuePropagator(program.Architecture, ssa);
 
+                    var icrw = new IndirectCallRewriter(program, ssa, eventListener);
+                    while (!eventListener.IsCanceled() && icrw.Rewrite())
+                    {
+                        vp.Transform();
+                        sst.RenameFrameAccesses = true;
```
Comment (uxmal) [inline src/Decompiler/Analysis/DataFlowAnalysis.cs]: Consider moving this out if the loop; it's not necessary to reset the property once it is set.

File: src/Decompiler/Analysis/IndirectCallRewriter.cs [hunk 2]
```diff
@@ -54,11 +58,13 @@ public class IndirectCallRewriter
             this.expander = new IndirectCallExpander(ssa);
             this.ssaIdTransformer = new SsaIdentifierTransformer(ssa);
             this.eventListener = eventListener;
+            this.emiter = new ExpressionEmitter();
```
Comment (uxmal) [inline src/Decompiler/Analysis/IndirectCallRewriter.cs]: It's spelled `emitter`; I'm considering switching from `emitter` to `m` everywhere, to be consistent with over 1000 unit tests that use `m`. Any thoughts?

File: src/Decompiler/Analysis/IndirectCallRewriter.cs [hunk 3]
```diff
@@ -54,11 +58,13 @@ public class IndirectCallRewriter
...
         }
 
-        public void Rewrite()
+        public bool Rewrite()
```
Comment (uxmal) [inline src/Decompiler/Analysis/IndirectCallRewriter.cs]: Please add a summary comment explaining what is being rewritten, and what it means when true/false is returned.

File: src/Decompiler/Analysis/IndirectCallRewriter.cs (hunk 2 above)
Comment (ptomin) [inline src/Decompiler/Analysis/IndirectCallRewriter.cs]: Done in 2966494

File: src/Decompiler/Analysis/DataFlowAnalysis.cs (hunk 1 above)
Comment (ptomin) [inline src/Decompiler/Analysis/DataFlowAnalysis.cs]: Done in 58d28bd

**Comment by ptomin** [comment]
I have done suggested fixes. Anything else, John?

**Comment by uxmal** [comment]
Looks good, Pavel.

**Pull request merged**